sensor/kubernetes/listener/resources/convert.go

```diff
@@ -348,7 +348,7 @@ func (w *deploymentWrap) getLabelSelector(spec reflect.Value) (*metav1.LabelSele
 		return nil, nil
 	}
 
-	// Selector is of map type for replication controller
+	// selector is of map type for replication controller
```

Comment:
Isn't it dangerous to have the function signature Match(label labels.Labels) but the labels parameter _has to be_ a labelWrap otherwise our cast labels.(labelWrap) fails? To me this seems like an easy source of errors. The labelWrap idea itself is fine, but if I go with it I would prefer to put labelWrap as a parameter of Matches, so "Matches(labels labelWrap)"

I understand that your approach avoids having to touch every place in the code where selectorWrapper.Matches is used, but in my estimation it's not worth the tradeoff.

Reply:
And good catch on that comment, I refactored Selector to be selector but it changed a few places that shouldn't have been changed.

Reply:
This was just a quick example. I agree, the idea of passing `labelWrap` (or Pitor's `LabelsHavingLength`) would be better 🙂